mesonbuild/backend/ninjabackend.py

```diff
@@ -991,10 +994,16 @@ def generate_cs_target(self, target, outfile):
         for rel_src in generated_sources.keys():
             dirpart, fnamepart = os.path.split(rel_src)
             if rel_src.lower().endswith('.cs'):
-                rel_srcs.append(rel_src)
-            deps.append(rel_src)
+                rel_srcs.append(os.path.normpath(rel_src))
+            deps.append(os.path.normpath(rel_src))
 
-        elem = NinjaBuildElement(self.all_outputs, outputs, 'cs_COMPILER', rel_srcs)
+        for dep in target.get_external_deps():
+            commands.extend_direct(dep.get_link_args())
+        commands += self.build.get_project_args(compiler, target.subproject)
+        commands += self.build.get_global_args(compiler)
+
+        elem = NinjaBuildElement(self.all_outputs, outputs, 'cs_COMPILER', rel_srcs,
+                allow_backslashes=True)
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->